lib/internal/http2/core.js

```diff
@@ -1844,7 +1871,31 @@ function createServer(options, handler) {
 // HTTP2-Settings header frame.
 function getPackedSettings(settings) {
   assertIsObject(settings, 'settings');
-  return binding.packSettings(settings || Object.create(null));
+  settings = Object.assign(Object.create(null), settings);
```

Comment:
AFAIK we are not mutating `settings` so cloning the object is not necessary. The previous `settings || Object.create(null)` seems preferable.